File: scripts/seed-students.ts
#!/usr/bin/env node
/**
 * Student Seeding Script for FlexiSched
 * Generates 1000 students with default password format: [firstName]@123
 */

import mongoose from 'mongoose';
import { MongoMemoryServer } from 'mongodb-memory-server';

// Student Schema (matching the existing schema)
const StudentSchema = new mongoose.Schema({
  _id: { type: String, required: true },
  studentId: { type: String, required: true, unique: true },
  firstName: { type: String, required: true },
  lastName: { type: String, required: true },
  email: { type: String, required: true, unique: true },
  password: { type: String, required: true }, // Add password field
  phone: { type: String, default: null },
  program: { type: String, required: true },
  semester: { type: Number, required: true },
  batch: { type: String, required: true },
  sectionId: { type: String, default: null },
  enrolledCourses: [{ type: String }],
  preferences: { type: Object, default: {} },
  isActive: { type: Boolean, default: true },
  createdAt: { type: Date, default: Date.now }
});

// Authentication Schema for login purposes
const AuthUserSchema = new mongoose.Schema({
  _id: { type: String, required: true },
  username: { type: String, required: true, unique: true },
  password: { type: String, required: true },
  role: { type: String, enum: ['admin', 'faculty', 'student'], required: true },
  userId: { type: String, required: true },
  email: { type: String, required: true },
  name: { type: String, required: true },
  isActive: { type: Boolean, default: true },
  createdAt: { type: Date, default: Date.now }
});

const StudentModel = mongoose.model('Student', StudentSchema);
const AuthUserModel = mongoose.model('AuthUser', AuthUserSchema);

interface StudentData {
  _id: string;
  studentId: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  plainPassword: string;
  phone: string;
  program: string;
  semester: number;
  batch: string;
  sectionId: string | null;
  enrolledCourses: string[];
  preferences: Record<string, any>;
  isActive: boolean;
  createdAt: Date;
}

class StudentSeeder {
  private mongoServer: MongoMemoryServer | null = null;

  async connect(): Promise<boolean> {
    try {
      // Check if already connected
      if (mongoose.connection.readyState === 1) {
        console.log('✅ Already connected to MongoDB');
        return true;
      }

      let uri = process.env.MONGODB_URI;
      
      // If no URI provided, use in-memory MongoDB
      if (!uri && !this.mongoServer) {
        console.log('🔄 No MONGODB_URI found, starting in-memory MongoDB server...');
        this.mongoServer = await MongoMemoryServer.create({
          instance: {
            dbName: 'timetable_system',
          },
        });
        uri = this.mongoServer.getUri();
        console.log('✅ In-memory MongoDB server started');
      } else if (!uri && this.mongoServer) {
        uri = this.mongoServer.getUri();
      }

      if (!uri) {
        throw new Error('No MongoDB URI available');
      }

      await mongoose.connect(uri, {
        serverSelectionTimeoutMS: 5000,
        connectTimeoutMS: 10000,
      });
      
      console.log('✅ Connected to MongoDB successfully');
      return true;
    } catch (error) {
      console.error('❌ MongoDB connection error:', error);
      return false;
    }
  }

  async disconnect(): Promise<void> {
    try {
      await mongoose.disconnect();
      if (this.mongoServer) {
        await this.mongoServer.stop();
        this.mongoServer = null;
        console.log('🔌 In-memory MongoDB server stopped');
      }
    } catch (error) {
      console.error('⚠️  Error disconnecting from MongoDB:', error);
    }
  }

  private async hashPassword(password: string): Promise<string> {
    // Simple hash for demo purposes - in production use proper bcrypt
    return `hashed_${password}`;
  }

  private generateStudentData(count: number = 1000): StudentData[] {
    // Expanded lists of Indian names for more variety
    const firstNamesMale = [
      'Aarav', 'Vivaan', 'Aditya', 'Vihaan', 'Arjun', 'Sai', 'Reyansh', 'Ayaan', 'Krishna', 'Ishaan',
      'Shaurya', 'Atharv', 'Advik', 'Pranav', 'Rishabh', 'Gagan', 'Arnav', 'Hriday', 'Kabir', 'Shivansh',
      'Rudra', 'Yuvraj', 'Dhruv', 'Karan', 'Aryan', 'Rohan', 'Ansh', 'Harsh', 'Dev', 'Kartik',
      'Nikhil', 'Rahul', 'Amit', 'Suresh', 'Rajesh', 'Vikram', 'Manoj', 'Sandeep', 'Deepak', 'Ashok',
      'Ravi', 'Anil', 'Sunil', 'Vinod', 'Prakash', 'Ramesh', 'Mahesh', 'Dinesh', 'Mukesh', 'Naresh'
    ];
    
    const firstNamesFemale = [
      'Ananya', 'Diya', 'Priya', 'Kavya', 'Aanya', 'Pari', 'Ira', 'Myra', 'Sara', 'Navya',
      'Aadya', 'Kiara', 'Saanvi', 'Avni', 'Riya', 'Ishika', 'Shanaya', 'Aditi', 'Vanya', 'Tara',
      'Aadhya', 'Arya', 'Siya', 'Nisha', 'Pooja', 'Sneha', 'Ritika', 'Nikita', 'Preeti', 'Swati',
      'Sunita', 'Meera', 'Kavita', 'Rekha', 'Geeta', 'Sushma', 'Vandana', 'Nisha', 'Divya', 'Shruti',
      'Ritu', 'Neha', 'Anjali', 'Sonia', 'Manisha', 'Kiran', 'Seema', 'Reena', 'Veena', 'Leela'
    ];
    
    const lastNames = [
      'Sharma', 'Verma', 'Gupta', 'Singh', 'Kumar', 'Jain', 'Agarwal', 'Patel', 'Shah', 'Mehta',
      'Joshi', 'Mishra', 'Tiwari', 'Yadav', 'Shukla', 'Pandey', 'Saxena', 'Srivastava', 'Dubey', 'Bansal',
      'Agrawal', 'Malhotra', 'Kapoor', 'Chopra', 'Arora', 'Sethi', 'Khanna', 'Bhatia', 'Tandon', 'Goel',
      'Mittal', 'Singhal', 'Goyal', 'Jindal', 'Bajaj', 'Saini', 'Khurana', 'Sachdeva', 'Bhalla', 'Sood',
      'Reddy', 'Rao', 'Nair', 'Iyer', 'Menon', 'Pillai', 'Das', 'Ghosh', 'Mukherjee', 'Chatterjee'
    ];
    
    const programs = [
      'Computer Science and Engineering',
      'Information Technology', 
      'Electronics and Communication Engineering',
      'Mechanical Engineering',
      'Civil Engineering',
      'Electrical Engineering',
      'Chemical Engineering',
      'Biotechnology',
      'Aerospace Engineering',
      'Automobile Engineering'
    ];
    
    const batches = ['2021', '2022', '2023', '2024', '2025'];
    
    const students: StudentData[] = [];
    const usedEmails = new Set<string>();
    const usedStudentIds = new Set<string>();
    
    console.log(`🔄 Generating ${count} student records...`);
    
    for (let i = 0; i < count; i++) {
      // Randomly select gender and corresponding first name
      const isMale = Math.random() > 0.5;
      const firstName = isMale 
        ? firstNamesMale[Math.floor(Math.random() * firstNamesMale.length)]
        : firstNamesFemale[Math.floor(Math.random() * firstNamesFemale.length)];
      const lastName = lastNames[Math.floor(Math.random() * lastNames.length)];
      
      // Generate unique student ID
      let studentId: string;
      do {
        studentId = `STU${Math.floor(Math.random() * 90000) + 10000}`;
      } while (usedStudentIds.has(studentId));
      usedStudentIds.add(studentId);
      
      // Generate unique email
      let email = `${firstName.toLowerCase()}.${lastName.toLowerCase()}@student.university.edu`;
      let counter = 1;
      while (usedEmails.has(email)) {
        email = `${firstName.toLowerCase()}.${lastName.toLowerCase()}${counter}@student.university.edu`;
        counter++;
      }
      usedEmails.add(email);
      
      // Generate password in format [firstName]@123
      const plainPassword = `${firstName}@123`;
      
      // Random program and semester
      const program = programs[Math.floor(Math.random() * programs.length)];
      const semester = Math.floor(Math.random() * 8) + 1;
      const batch = batches[Math.floor(Math.random() * batches.length)];
      
      // Generate phone number
      const phone = `+91${Math.floor(Math.random() * 3000000000) + 7000000000}`;
      
      const student: StudentData = {
        _id: `student-${i + 1001}`, // Start from 1001 to avoid conflicts
        studentId,
        firstName,
        lastName,
        email,
        password: '', // Will be hashed later
        plainPassword,
        phone,
        program,
        semester,
        batch: `Batch-${batch}`,
        sectionId: null,
        enrolledCourses: [],
        preferences: {},
        isActive: true,
        createdAt: new Date()
      };
      
      students.push(student);
      
      // Progress indicator
      if ((i + 1) % 100 === 0) {
        console.log(`   Generated ${i + 1}/${count} students...`);
      }
    }
    
    console.log(`✅ Generated ${students.length} student records`);
    return students;
  }

  private async clearExistingStudents(confirm: boolean = false): Promise<boolean> {
    if (!confirm) {
      // In a real CLI, you'd use readline, but for simplicity we'll skip confirmation
      console.log('⚠️  Clearing existing students...');
    }
    
    try {
      const studentResult = await StudentModel.deleteMany({});
      const authResult = await AuthUserModel.deleteMany({ role: 'student' });
      console.log(`🗑️  Deleted ${studentResult.deletedCount} existing students and ${authResult.deletedCount} auth records`);
      return true;
    } catch (error) {
      console.error('❌ Error clearing students:', error);
      return false;
    }
  }

  private async insertStudents(students: StudentData[], batchSize: number = 100): Promise<boolean> {
    try {
      const total = students.length;
      console.log(`📥 Hashing passwords and inserting ${total} students in batches of ${batchSize}...`);
      
      // Hash all passwords first
      for (let i = 0; i < students.length; i++) {
        students[i].password = await this.hashPassword(students[i].plainPassword);
        if ((i + 1) % 100 === 0) {
          console.log(`   Hashed passwords for ${i + 1}/${total} students...`);
        }
      }
      
      // Insert in batches
      for (let i = 0; i < total; i += batchSize) {
        const batch = students.slice(i, i + batchSize);
        const studentsToInsert = batch.map(s => ({
          _id: s._id,
          studentId: s.studentId,
          firstName: s.firstName,
          lastName: s.lastName,
          email: s.email,
          password: s.password,
          phone: s.phone,
          program: s.program,
          semester: s.semester,
          batch: s.batch,
          sectionId: s.sectionId,
          enrolledCourses: s.enrolledCourses,
          preferences: s.preferences,
          isActive: s.isActive,
          createdAt: s.createdAt
        }));
        
        await StudentModel.insertMany(studentsToInsert);
        console.log(`   Inserted batch ${Math.floor(i/batchSize) + 1}/${Math.ceil(total/batchSize)}`);
      }
      
      console.log(`✅ Successfully inserted ${total} students`);
      return true;
    } catch (error) {
      console.error('❌ Error inserting students:', error);
      return false;
    }
  }

  private async createAuthenticationRecords(students: StudentData[]): Promise<boolean> {
    try {
      const authRecords = students.map(student => ({
        _id: `auth-${student.studentId}`,
        username: student.studentId,
        password: student.password,
        role: 'student' as const,
        userId: student._id,
        email: student.email,
        name: `${student.firstName} ${student.lastName}`,
        isActive: true,
        createdAt: new Date()
      }));
      
      await AuthUserModel.insertMany(authRecords);
      console.log(`✅ Created ${authRecords.length} authentication records`);
      return true;
    } catch (error) {
      console.error('❌ Error creating authentication records:', error);
      return false;
    }
  }

  private async generatePasswordReport(students: StudentData[], outputFile: string = 'student_passwords.txt'): Promise<boolean> {
    try {
      const fs = await import('fs');
      let content = 'FlexiSched - Student Login Credentials\n';
      content += '='.repeat(50) + '\n\n';
      content += 'Format: Student ID | Name | Email | Password\n';
      content += '-'.repeat(80) + '\n';
      
      for (const student of students) {
        content += `${student.studentId} | ${student.firstName} ${student.lastName} | ${student.email} | ${student.plainPassword}\n`;
      }
      
      content += `\nTotal Students: ${students.length}\n`;
      content += `Generated on: ${new Date().toISOString()}\n`;
      
      fs.writeFileSync(outputFile, content, 'utf-8');
      console.log(`📄 Password report saved to: ${outputFile}`);
      return true;
    } catch (error) {
      console.error('❌ Error generating password report:', error);
      return false;
    }
  }

  async seedStudents(count: number = 1000, clearExisting: boolean = false): Promise<boolean> {
    console.log('🌱 Starting student seeding process...');
    console.log(`   Target count: ${count}`);
    console.log(`   Clear existing: ${clearExisting}`);
    
    // Connect to database
    if (!await this.connect()) {
      return false;
    }
    
    try {
      // Clear existing students if requested
      if (clearExisting) {
        if (!await this.clearExistingStudents(true)) {
          return false;
        }
      }
      
      // Generate student data
      const students = this.generateStudentData(count);
      
      // Insert students
      if (!await this.insertStudents(students)) {
        return false;
      }
      
      // Create authentication records
      if (!await this.createAuthenticationRecords(students)) {
        console.log('⚠️  Warning: Failed to create authentication records');
      }
      
      // Generate password report
      if (!await this.generatePasswordReport(students)) {
        console.log('⚠️  Warning: Failed to generate password report');
      }
      
      console.log(`🎉 Successfully seeded ${count} students!`);
      console.log(`   Password format: [firstName]@123`);
      console.log(`   Example: Aarav@123, Priya@123, etc.`);
      
      return true;
      
    } catch (error) {
      console.error('❌ Seeding failed:', error);
      return false;
    } finally {
      await this.disconnect();
    }
  }
}

// Main execution
async function main() {
  const args = process.argv.slice(2);
  const count = args.includes('--count') ? parseInt(args[args.indexOf('--count') + 1]) || 1000 : 1000;
  const clearExisting = args.includes('--clear');
  
  console.log('🚀 FlexiSched Student Seeder');
  console.log('============================');
  
  const seeder = new StudentSeeder();
  const success = await seeder.seedStudents(count, clearExisting);
  
  if (success) {
    console.log('\n✅ Seeding completed successfully!');
    console.log('📋 Check \'student_passwords.txt\' for login credentials');
    process.exit(0);
  } else {
    console.log('\n❌ Seeding failed!');
    process.exit(1);
  }
}

// Run if this file is executed directly
if (import.meta.url === `file://${process.argv[1]}`) {
  main().catch(console.error);
}

export { StudentSeeder };

File: scripts/seed_students.py
#!/usr/bin/env python3
"""
Student Seeding Script for FlexiSched
Generates 1000 students with default password format: [firstName]@123
"""

import asyncio
import sys
import os
import random
from datetime import datetime
from typing import List, Dict

# Add the parent directory to the Python path to import from ai_engine
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

# Import MongoDB connection utilities
try:
    import pymongo
    from pymongo import MongoClient
    import bcrypt
except ImportError:
    print("Required packages not installed. Please install: pip install pymongo bcrypt")
    sys.exit(1)

class StudentSeeder:
    def __init__(self, mongodb_uri: str = None):
        """Initialize the seeder with MongoDB connection"""
        self.mongodb_uri = mongodb_uri or os.getenv('MONGODB_URI', 'mongodb://localhost:27017/timetable')
        self.client = None
        self.db = None
        
    def connect(self):
        """Connect to MongoDB"""
        try:
            self.client = MongoClient(self.mongodb_uri)
            # Extract database name from URI or use default
            if 'timetable_system' in self.mongodb_uri:
                self.db = self.client['timetable_system']
            else:
                self.db = self.client['timetable_system']
            
            # Test connection
            self.client.admin.command('ping')
            print(f"✅ Connected to MongoDB: {self.mongodb_uri}")
            return True
        except Exception as e:
            print(f"❌ Failed to connect to MongoDB: {e}")
            return False
    
    def disconnect(self):
        """Disconnect from MongoDB"""
        if self.client:
            self.client.close()
            print("🔌 Disconnected from MongoDB")
    
    def hash_password(self, password: str) -> str:
        """Hash password using bcrypt"""
        salt = bcrypt.gensalt()
        hashed = bcrypt.hashpw(password.encode('utf-8'), salt)
        return hashed.decode('utf-8')
    
    def generate_student_data(self, count: int = 1000) -> List[Dict]:
        """Generate student data with realistic Indian names"""
        
        # Expanded lists of Indian names for more variety
        first_names_male = [
            'Aarav', 'Vivaan', 'Aditya', 'Vihaan', 'Arjun', 'Sai', 'Reyansh', 'Ayaan', 'Krishna', 'Ishaan',
            'Shaurya', 'Atharv', 'Advik', 'Pranav', 'Rishabh', 'Gagan', 'Arnav', 'Hriday', 'Kabir', 'Shivansh',
            'Rudra', 'Yuvraj', 'Dhruv', 'Karan', 'Aryan', 'Rohan', 'Ansh', 'Harsh', 'Dev', 'Kartik',
            'Nikhil', 'Rahul', 'Amit', 'Suresh', 'Rajesh', 'Vikram', 'Manoj', 'Sandeep', 'Deepak', 'Ashok',
            'Ravi', 'Anil', 'Sunil', 'Vinod', 'Prakash', 'Ramesh', 'Mahesh', 'Dinesh', 'Mukesh', 'Naresh'
        ]
        
        first_names_female = [
            'Ananya', 'Diya', 'Priya', 'Kavya', 'Aanya', 'Pari', 'Ira', 'Myra', 'Sara', 'Navya',
            'Aadya', 'Kiara', 'Saanvi', 'Avni', 'Riya', 'Ishika', 'Shanaya', 'Aditi', 'Vanya', 'Tara',
            'Aadhya', 'Arya', 'Siya', 'Nisha', 'Pooja', 'Sneha', 'Ritika', 'Nikita', 'Preeti', 'Swati',
            'Sunita', 'Meera', 'Kavita', 'Rekha', 'Geeta', 'Sushma', 'Vandana', 'Nisha', 'Divya', 'Shruti',
            'Ritu', 'Neha', 'Anjali', 'Sonia', 'Manisha', 'Kiran', 'Seema', 'Reena', 'Veena', 'Leela'
        ]
        
        last_names = [
            'Sharma', 'Verma', 'Gupta', 'Singh', 'Kumar', 'Jain', 'Agarwal', 'Patel', 'Shah', 'Mehta',
            'Joshi', 'Mishra', 'Tiwari', 'Yadav', 'Shukla', 'Pandey', 'Saxena', 'Srivastava', 'Dubey', 'Bansal',
            'Agrawal', 'Malhotra', 'Kapoor', 'Chopra', 'Arora', 'Sethi', 'Khanna', 'Bhatia', 'Tandon', 'Goel',
            'Mittal', 'Singhal', 'Goyal', 'Jindal', 'Bajaj', 'Saini', 'Khurana', 'Sachdeva', 'Bhalla', 'Sood',
            'Reddy', 'Rao', 'Nair', 'Iyer', 'Menon', 'Pillai', 'Das', 'Ghosh', 'Mukherjee', 'Chatterjee'
        ]
        
        programs = [
            'Computer Science and Engineering',
            'Information Technology', 
            'Electronics and Communication Engineering',
            'Mechanical Engineering',
            'Civil Engineering',
            'Electrical Engineering',
            'Chemical Engineering',
            'Biotechnology',
            'Aerospace Engineering',
            'Automobile Engineering'
        ]
        
        batches = ['2021', '2022', '2023', '2024', '2025']
        
        students = []
        used_emails = set()
        used_student_ids = set()
        
        print(f"🔄 Generating {count} student records...")
        
        for i in range(count):
            # Randomly select gender and corresponding first name
            is_male = random.choice([True, False])
            first_name = random.choice(first_names_male if is_male else first_names_female)
            last_name = random.choice(last_names)
            
            # Generate unique student ID
            while True:
                student_id = f"STU{random.randint(10000, 99999)}"
                if student_id not in used_student_ids:
                    used_student_ids.add(student_id)
                    break
            
            # Generate unique email
            base_email = f"{first_name.lower()}.{last_name.lower()}"
            email = f"{base_email}@student.university.edu"
            counter = 1
            while email in used_emails:
                email = f"{base_email}{counter}@student.university.edu"
                counter += 1
            used_emails.add(email)
            
            # Generate password in format [firstName]@123
            password = f"{first_name}@123"
            hashed_password = self.hash_password(password)
            
            # Random program and semester
            program = random.choice(programs)
            semester = random.randint(1, 8)
            batch = random.choice(batches)
            
            # Generate phone number
            phone = f"+91{random.randint(7000000000, 9999999999)}"
            
            student = {
                '_id': f"student-{i + 1001}",  # Start from 1001 to avoid conflicts
                'studentId': student_id,
                'firstName': first_name,
                'lastName': last_name,
                'email': email,
                'password': hashed_password,  # Store hashed password
                'plainPassword': password,    # Store plain password for reference (remove in production)
                'phone': phone,
                'program': program,
                'semester': semester,
                'batch': f"Batch-{batch}",
                'sectionId': None,
                'enrolledCourses': [],
                'preferences': {},
                'isActive': True,
                'createdAt': datetime.now()
            }
            
            students.append(student)
            
            # Progress indicator
            if (i + 1) % 100 == 0:
                print(f"   Generated {i + 1}/{count} students...")
        
        print(f"✅ Generated {len(students)} student records")
        return students
    
    def clear_existing_students(self, confirm: bool = False):
        """Clear existing students and authentication records (use with caution)"""
        if not confirm:
            response = input("⚠️  This will delete ALL existing students. Are you sure? (yes/no): ")
            if response.lower() != 'yes':
                print("❌ Operation cancelled")
                return False
        
        try:
            # Clear students collection
            result = self.db.students.delete_many({})
            print(f"🗑️  Deleted {result.deleted_count} existing students")
            
            # Clear authentication records for students
            auth_result = self.db.auth_users.delete_many({"role": "student"})
            print(f"🗑️  Deleted {auth_result.deleted_count} existing student authentication records")
            
            return True
        except Exception as e:
            print(f"❌ Error clearing students: {e}")
            return False
    
    def insert_students(self, students: List[Dict], batch_size: int = 100):
        """Insert students in batches"""
        try:
            total = len(students)
            print(f"📥 Inserting {total} students in batches of {batch_size}...")
            
            for i in range(0, total, batch_size):
                batch = students[i:i + batch_size]
                self.db.students.insert_many(batch)
                print(f"   Inserted batch {i//batch_size + 1}/{(total-1)//batch_size + 1}")
            
            print(f"✅ Successfully inserted {total} students")
            return True
        except Exception as e:
            print(f"❌ Error inserting students: {e}")
            return False
    
    def create_authentication_collection(self, students: List[Dict]):
        """Create a separate authentication collection for login purposes"""
        try:
            auth_records = []
            for student in students:
                auth_record = {
                    '_id': f"auth-{student['studentId']}",
                    'username': student['studentId'],
                    'password': student['password'],
                    'role': 'student',
                    'userId': student['_id'],
                    'email': student['email'],
                    'name': f"{student['firstName']} {student['lastName']}",
                    'isActive': True,
                    'createdAt': datetime.now()
                }
                auth_records.append(auth_record)
            
            # Insert authentication records
            self.db.auth_users.insert_many(auth_records)
            print(f"✅ Created {len(auth_records)} authentication records")
            return True
        except Exception as e:
            print(f"❌ Error creating authentication records: {e}")
            return False
    
    def generate_password_report(self, students: List[Dict], output_file: str = "student_passwords.txt"):
        """Generate a report of all student passwords for reference"""
        try:
            with open(output_file, 'w', encoding='utf-8') as f:
                f.write("FlexiSched - Student Login Credentials\n")
                f.write("=" * 50 + "\n\n")
                f.write("Format: Student ID | Name | Email | Password\n")
                f.write("-" * 80 + "\n")
                
                for student in students:
                    f.write(f"{student['studentId']} | {student['firstName']} {student['lastName']} | {student['email']} | {student['plainPassword']}\n")
                
                f.write(f"\nTotal Students: {len(students)}\n")
                f.write(f"Generated on: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}\n")
            
            print(f"📄 Password report saved to: {output_file}")
            return True
        except Exception as e:
            print(f"❌ Error generating password report: {e}")
            return False
    
    def seed_students(self, count: int = 1000, clear_existing: bool = False):
        """Main seeding function"""
        print(f"🌱 Starting student seeding process...")
        print(f"   Target count: {count}")
        print(f"   Clear existing: {clear_existing}")
        
        # Connect to database
        if not self.connect():
            return False
        
        try:
            # Clear existing students if requested
            if clear_existing:
                if not self.clear_existing_students():
                    return False
            
            # Generate student data
            students = self.generate_student_data(count)
            
            # Insert students
            if not self.insert_students(students):
                return False
            
            # Create authentication records
            if not self.create_authentication_collection(students):
                print("⚠️  Warning: Failed to create authentication records")
            
            # Generate password report
            if not self.generate_password_report(students):
                print("⚠️  Warning: Failed to generate password report")
            
            print(f"🎉 Successfully seeded {count} students!")
            print(f"   Password format: [firstName]@123")
            print(f"   Example: Aarav@123, Priya@123, etc.")
            
            return True
            
        except Exception as e:
            print(f"❌ Seeding failed: {e}")
            return False
        finally:
            self.disconnect()

def main():
    """Main function with command line interface"""
    import argparse
    
    parser = argparse.ArgumentParser(description='Seed FlexiSched with student data')
    parser.add_argument('--count', type=int, default=1000, help='Number of students to create (default: 1000)')
    parser.add_argument('--clear', action='store_true', help='Clear existing students before seeding')
    parser.add_argument('--mongodb-uri', type=str, help='MongoDB connection URI')
    
    args = parser.parse_args()
    
    # Create seeder instance
    seeder = StudentSeeder(args.mongodb_uri)
    
    # Run seeding
    success = seeder.seed_students(
        count=args.count,
        clear_existing=args.clear
    )
    
    if success:
        print("\n✅ Seeding completed successfully!")
        print("📋 Check 'student_passwords.txt' for login credentials")
    else:
        print("\n❌ Seeding failed!")
        sys.exit(1)

if __name__ == "__main__":
    main()
